modules/core/shared/src/main/scala/util/Pool.scala

```diff
@@ -152,7 +152,7 @@ object Pool {
           ref.modify {
             case (os, Nil) =>  ((os :+ None, Nil), ().pure[F]) // new empty slot
             case (os, d :: ds) =>  ((os, ds), Concurrent[F].attempt(rsrc(Tracer[F]).allocated).flatMap(d.complete).void) // alloc now!
-          } .guarantee(a._2).flatten
+          }.flatten.guarantee(a._2)
```

Comment:
Upon losing connection to the server, this `dispose` is called. The deferred removed here (`d`) should be completed but it's not b/c the finalizer of `A` completes with the connection error.